app/models/trail.rb

```diff
@@ -1,4 +1,16 @@
 class Trail < ActiveRecord::Base
   has_many :activities
   has_many :species
+
+  scope :filter_by_state, -> (state) {
+    where("lower(state) LIKE ?", "%#{state.downcase}%")
+  }
+
+  scope :filter_by_lat, -> (lat) {
+    where("lat BETWEEN ? AND ?", lat - 1, lat + 1)
```

Comment:
I still wish we figured out how to return the trails in order of lat & long proximity